core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java

```diff
@@ -208,7 +208,8 @@ public void start(WorkerNetAddress address) throws IOException {
 
     // Setup BlockMasterSync
     BlockMasterSync blockMasterSync = mResourceCloser
-        .register(new BlockMasterSync(this, mWorkerId, mAddress, mBlockMasterClientPool));
+        .register(new BlockMasterSync(
+            this, mWorkerId, mAddress, mBlockMasterClientPool, mHeartbeatReporter));
```

Comment:
in the future, there might be more than one BlockMasterSync() so we pass in a dedicated reporter instead of using the one from the worker.